scripts/travis-run.sh

```diff
@@ -1,6 +1,16 @@
 #!/bin/bash
 set -euo pipefail
 
+if [[ $TRAVIS_BRANCH != "master" && $TRAVIS_PULL_REQUEST == "false" && $TRAVIS_REPO_SLUG == "bioconda/bioconda-recipes" ]]
+then
+    echo ""
+    echo "Tests are skipped for pushes to the main bioconda-recipes repo."
+    echo "If you have opened a pull request, please see the full tests for that PR."
+    echo "See https://bioconda.github.io/build-system.html for details"
+    echo ""
+    exit 0
+fi
+
```

Comment:
I do sometimes :). Actually I think the big use-case for this is testing on OSX for people without access to a Mac.

If we exit 0 on the branch push, I think the only disadvantage is that people might get confused when the full test on the PR fails but the branch push passes. That can be alleviated with informative messages and documentation.